libraries/standard/mqtt/utest/mqtt_lightweight_utest.c

```diff
@@ -0,0 +1,177 @@
+{
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    size_t remainingLength, packetSize;
+    MQTTFixedBuffer_t networkBuffer;
+    MQTTConnectInfo_t connectInfo;
+
+    /* Test NULL pConnectInfo. */
+    mqttStatus = MQTT_SerializeConnect( NULL, NULL,
+                                        remainingLength, &networkBuffer );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    /* Test NULL pBuffer. */
+    mqttStatus = MQTT_SerializeConnect( &connectInfo, NULL,
+                                        remainingLength, NULL );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    /* Test connectPacketSize > pBuffer->size. */
+    /* Get MQTT connect packet size and remaining length. */
+    mqttStatus = MQTT_GetConnectPacketSize( &connectInfo,
+                                            NULL,
+                                            &remainingLength,
+                                            &packetSize );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    networkBuffer.pBuffer = mqttBuffer;
+    networkBuffer.size = remainingLength - 1;
+    mqttStatus = MQTT_SerializeConnect( &connectInfo, NULL,
+                                        remainingLength, &networkBuffer );
+    TEST_ASSERT_EQUAL( MQTTNoMemory, mqttStatus );
+}
+
+/**
+ * @brief This method calls Mqtt_SerializeConnect successfully using different parameters
+ * until we have full coverage on the private method, serializeConnectPacket(...).
+ */
+void test_Mqtt_SerializeConnect_happy_path()
+{
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    size_t remainingLength, packetSize;
+    MQTTFixedBuffer_t networkBuffer;
+    MQTTConnectInfo_t connectInfo;
+    MQTTPublishInfo_t willInfo;
+
+    /* Fill structs to pass into methods to be tested. */
+    setupNetworkBuffer( &networkBuffer );
+    setupConnectInfo( &connectInfo );
+    setupWillInfo( &willInfo );
+
+    /* Get MQTT connect packet size and remaining length. */
+    mqttStatus = MQTT_GetConnectPacketSize( &connectInfo,
+                                            &willInfo,
+                                            &remainingLength,
+                                            &packetSize );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    mqttStatus = MQTT_SerializeConnect( &connectInfo, &willInfo,
+                                        remainingLength, &networkBuffer );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+
+    /* Repeat with QoS2. */
+    willInfo.qos = MQTTQoS2;
+    mqttStatus = MQTT_GetConnectPacketSize( &connectInfo,
+                                            &willInfo,
+                                            &remainingLength,
+                                            &packetSize );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    mqttStatus = MQTT_SerializeConnect( &connectInfo, &willInfo,
+                                        remainingLength, &networkBuffer );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
```

Comment:
I think it would make a good addition to validate the serialized buffer values as well here.